libr/main/rabin2.c

```diff
@@ -471,6 +471,48 @@ static int rabin_show_srcline(RBin *bin, ut64 at) {
 	return false;
 }
 
+/* Map Sections to Segments https://github.com/radareorg/radare2/issues/14647 */
+static int rabin_map_sections_to_segments(RBin *bin){
+	RListIter *iter;
+	RBinSection *section = NULL, *segment = NULL;
+	RList *sections = r_list_new();
+	RList *segments = r_list_new();
+	RList *tmp = r_bin_get_sections (bin);
+	printf ("Section to Segment mapping:\n");
+	printf ("Segment\tSections...\n");	
+
+	r_list_foreach(tmp, iter, section) {
+		if (section->is_segment) {
+			r_list_append (segments, section);
+		} else {
+			r_list_append (sections, section);
+		}
+	}
+
+	segment = r_list_pop_head (segments);
+	RInterval segment_itv = (RInterval){segment->vaddr, segment->size};
+	printf ("%s\t", segment->name);
+
+	r_list_foreach (sections, iter, section) {
+		
+		if (r_itv_contain (segment_itv, section->vaddr)) {
+			printf ("%s ", section->name);
+		} else {
+			segment = r_list_pop_head (segments);
+			segment_itv = (RInterval){segment->vaddr, segment->size};
+			printf ("\n");
+			printf ("%s\t", segment->name);
+		}
+	}
+	printf ("\n");
+	/* print out left-over segments that contains no sections */
+	while (r_list_length(segments) > 0) {
+		segment = r_list_pop_head (segments);
+		printf ("%s\t\n", segment->name);
```

Comment:
use table api instead of printfs with tabs or at least %20s and such, but tabs are not reliable for consistent output across editors/visualizers/shells/fonts/.. its better not to use them